volumes/vds/py/vds_functions.py

```diff
@@ -0,0 +1,352 @@
+#pull the detector locations table (`entitylocations`) from ITS Central and insert new rows into RDS `vds.entity_locations`.
+#very small table so OK to pull entire table daily. 
+
+    from psycopg2 import sql, Error
+    from psycopg2.extras import execute_values
+
+    # Pull data from the detector_inventory table
+    entitylocation_sql = sql.SQL('''
+    SELECT 
+         divisionid,
+         entitytype,
+         entityid,
+         TIMEZONE('UTC', locationtimestamputc) AT TIME ZONE 'EST5EDT' AS locationtimestamp,
+         latitude,
+         longitude,
+         altitudemetersasl,
+         headingdegrees,
+         speedkmh,
+         numsatellites,
+         dilutionofprecision,
+         mainroadid,
+         crossroadid,
+         secondcrossroadid,
+         mainroadname,
+         crossroadname,
+         secondcrossroadname,
+         streetnumber,
+         offsetdistancemeters,
+         offsetdirectiondegrees,
+         locationsource,
+         locationdescriptionoverwrite
+    FROM public.entitylocation
+    WHERE divisionid IN (2, 8001) --only these have data in 'vdsdata' table
+    ''')
+
+    try:
+        with itsc_conn.get_conn() as con:
+            with con.cursor() as cur:
+                cur.execute(entitylocation_sql)
+                entitylocations = cur.fetchall()
+                LOGGER.info('Fetching entitylocation')
+    except Error as exc:
+        LOGGER.critical('Error fetching entitylocation.')
+        LOGGER.critical(exc)
+        con.close()
+
+    LOGGER.info(f"Number of rows fetched from entitylocations table: {len(entitylocations)}")
+
+    # upsert data
+    upsert_query = sql.SQL('''
+        INSERT INTO vds.vds_entity_locations (
+            divisionid, entitytype, entityid, locationtimestamp, latitude, longitude, altitudemetersasl,
+            headingdegrees, speedkmh, numsatellites, dilutionofprecision, mainroadid, crossroadid,
+            secondcrossroadid, mainroadname, crossroadname, secondcrossroadname, streetnumber,
+            offsetdistancemeters, offsetdirectiondegrees, locationsource, locationdescriptionoverwrite)
+        VALUES %s
+        ON CONFLICT DO NOTHING;
+    ''')
+
+    try:
+        with rds_conn.get_conn() as con:
+            with con.cursor() as cur:
+                execute_values(cur, upsert_query, entitylocations)
+                LOGGER.info('Inserting vds_entity_locations')
+    except Error as exc:
+        LOGGER.critical('Error inserting vds_entity_locations.')
+        LOGGER.critical(exc)
+        con.close()
+
+
+def parse_lane_data(laneData):
+# Parse binary vdsdata.lanedata column
+
+    import struct
+    result = []
+
+    with memoryview(laneData) as mv:
+        i = 0 #index within memoryview
+        while i < len(mv):
+            # Get lane
+            lane = mv[i][0]
+
+            # Get speed
+            #Stored in km/h * 100. 65535 for null value. Convert 0 to null to maintain backward compatibility
```

Comment:
Do you mean that the multiplier `100. 65535` is only applied on null values? Or is it applied to all records?